main.py

```diff
@@ -0,0 +1,404 @@
+
+        args:
+            filename: file to include, relative to the top directory of the documentation project.
+            start_line (Optional): if included, will start including the file from this line
+                (indexed to 0)
+            end_line (Optional): if included, will stop including at this line (indexed to 0)
+        """
+        full_filename = os.path.join(env.project_dir, filename)
+        with open(full_filename, "r") as f:
+            lines = f.readlines()
+        line_range = lines[start_line:end_line]
+        content = "".join(line_range)
+
+        _filenamebase = env.page.file.url
+
+        for _find, _replace in FIND_REPLACE.items():
+            if _filenamebase in _replace:
+                _replace = _replace.replace(_filenamebase, "")
+            content = content.replace(_find, _replace)
+        return content
+
+    @env.macro
+    def frictionless_spec(spec_filename: str = None) -> str:
+        if not base_path:
+            base_path = os.path.dirname(os.path.realpath(__file__))
+        if not docs_path:
+            docs_path = os.path.join(base_path, "docs")
+
+        if spec_filename is None:
+            spec_filename = glob.glob(
+                os.path.join(base_path, "**/*.spec.json"), recursive=True
+            )[0]
+
+        print("Documenting spec_file: ", spec_filename)
+
+        # Generate a table for overall file requirements
+        spec_df = read_config(spec_filename)
+        spec_df = spec_df.drop(
+            columns=["fullpath", "fullpath_schema", "path", "schema", "name"]
+        ).reset_index()
+        spec_df["name"] = spec_df["name"].apply(
+            lambda x: "[`{}`](tables.md#{})".format(x, x)
+        )
+
+        _fill_template(
+            outfile_path=os.path.join(docs_path, outfile_name),
+            content_dict={"SPEC": spec_df.to_markdown(index=False)},
+        )
+
+    @env.macro
+    def list_examples(data_dir: str) -> str:
+        """Outputs a simple list of the directories in a folder in markdown.
+
+        Args:
+            data_dir (str):directory to search in
+
+        Returns:
+            str: markdown-formatted list
+        """
+        data_dir = os.path.join(env.project_dir, data_dir)
+        examples = [
+            d for d in os.listdir(data_dir) if os.path.isdir(os.path.join(data_dir, d))
+        ]
+        sep = "\n- "
+        bullet_list = sep + sep.join([_to_md_link(e) for e in examples])
+
+        example_docs = bullet_list
+
+        for e in examples:
+            example_title = f"\n## {e.capitalize()}\n\n"
+            example_docs += example_title
+            _readme_filename = os.path.join(data_dir, e, "README.md")
+            if os.path.exists(_readme_filename):
+                example_docs += include_file(_readme_filename, start_line=2)
+            else:
+                example_docs += (
+                    f"No README.md in example folder {os.path.join(data_dir,e)}.\n"
+                )
```

Comment:
This could end up being very large if we have a couple dozen examples with readmes.

How about creating a table and linking to the readme if available? e.g.,

## Examples
| Name | Agency |Vendors| Description | Resources |
|-|-|-|-|-|
| [Example](data/example) | | |[readme](data/example/README.md) | devices, vehicle_locations, fare_transactions, train_cars, operators, stop_visits, vehicle_train_cars, vehicles, trips_performed, station_activities, passenger_events|
| [MetroTransit](data/50027-MetroTransitMN)|50027|TransitMaster, Cubic|| vehicle_locations|

Reply:
fixed